packages/govuk-frontend/src/govuk/common/index.mjs

```diff
@@ -149,19 +149,32 @@ export function isSupported($scope = document.body) {
 }
 
 /**
+ * Validates Component config against a schema
+ *
  * @protected
  * @param {Schema} schema
  * @param {Config[ConfigKey]} config
- * @returns {boolean} Whether the config is valid
+ * @returns {string[]} An array of errors, or an empty array if there are no errors
  */
-export function isConfigValid(schema, config) {
+export function validateConfig(schema, config) {
+  /**
+   * @type {string[]}
+   */
+  const errors = []
+
   if (!schema.anyOf) {
-    return true
+    return errors
   }
 
-  return schema.anyOf.some(({ required }) =>
+  const satisfiesAnyOf = schema.anyOf.some(({ required }) =>
     required.every((prop) => prop in config && config[prop])
   )
+
+  if (!satisfiesAnyOf) {
+    errors.push('Either `maxlength` or `maxwords` must be provided')
```

Comment:
Great minds, this is where I'd pluck that example

```suggestion
    errors.push(schema.anyOf.errorMessage)
```

Reply:
Pushed up this change.